src/cmd-push-container-manifest

```diff
@@ -60,16 +59,24 @@ def main():
             images.append(f"oci-archive:{ociarchive}")
 
         # Create/Upload the manifest list to the container registry
-        with tempfile.NamedTemporaryFile() as digestfile:
-            create_and_push_container_manifest(
-                args.repo, args.tag, images, args.v2s2, digestfile.name)
-            digestfile.seek(0)
-            digest = digestfile.read().decode('utf-8').strip()
+        manifest = create_and_push_container_manifest(
+            args.repo, args.tags, images, args.v2s2)
 
-        # Update the meta.json in each build/arch metadata
-        for _, buildmeta in buildmetas.items():
-            buildmeta[args.metajsonname] = {'image': f"{args.repo}@{digest}"}
-            buildmeta.write(artifact_name=args.metajsonname)
+        # Update the `meta.json` files. Note the digest included is the
+        # arch-specific one for each individual arch, and not the manifest list
+        # digest. See: https://github.com/coreos/coreos-assembler/issues/3122.
+        assert len(manifest['manifests']) == len(buildmetas)
+        for manifest in manifest['manifests']:
```

Comment:
I know this is probably safe because it's scoped but i think it would be prudent to use a different variable here OR above. My suggestion is to use:

```
manifest_list = create_and_push_container_manifest()
```

above.

Reply:
or `manifest_info` like is done in `container_manifest.py`

Reply:
Yeah good point. Renamed!